Tests/Common/Securities/SecurityIdentifierTests.cs

```diff
@@ -389,6 +391,284 @@ public void GenerateBase_UsesProvidedSymbol_WhenBaseDataTypeIsNull()
+            Assert.AreEqual(option.ID.OptionStyle, sid.OptionStyle);
+            Assert.AreEqual(option.ID.Market, sid.Market);
+            Assert.AreEqual(option.ID.SecurityType, sid.SecurityType);
+        }
+
+        [Test]
+        public void NegativeFractionalNumberStrikePriceApproachesBoundsWithoutOverflowingSid()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            var option = Symbol.CreateOption(
+                equity,
+                Market.USA,
+                OptionStyle.American,
+                OptionRight.Call,
+                -47.5711m,
+                new DateTime(2020, 5, 21));
+
+            // The SID specification states that the total width for the properties value
+            // is at most 20 digits long. If we overflowed the SID, the strike price can and will
+            // eat from other slots, corrupting the data. If we have no overflow, the SID will
+            // be constructed properly without corrupting any data, even as we approach the bounds.
+            // We will assert that all properties contained within the _properties field are valid and not corrupted
+            var sid = SecurityIdentifier.Parse(option.ID.ToString());
+
+            Assert.AreEqual(new DateTime(2020, 5, 21), sid.Date);
+            Assert.AreEqual(-47.5711m, sid.StrikePrice);
+            Assert.AreEqual(OptionRight.Call, sid.OptionRight);
+            Assert.AreEqual(OptionStyle.American, sid.OptionStyle);
+            Assert.AreEqual(Market.USA, sid.Market);
+            Assert.AreEqual(SecurityType.Option, sid.SecurityType);
+
+            Assert.AreEqual(option.ID.Date,sid.Date);
+            Assert.AreEqual(option.ID.StrikePrice, sid.StrikePrice);
+            Assert.AreEqual(option.ID.OptionRight, sid.OptionRight);
+            Assert.AreEqual(option.ID.OptionStyle, sid.OptionStyle);
+            Assert.AreEqual(option.ID.Market, sid.Market);
+            Assert.AreEqual(option.ID.SecurityType, sid.SecurityType);
+        }
+
+        [TestCase(475712.0)]
+        [TestCase(47.5712)]
+        [TestCase(999999.0)]
+        [TestCase(-475712.0)]
+        [TestCase(-47.5712)]
+        [TestCase(-999999)]
+        public void HighPrecisionNumberThrows(double strike)
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    (decimal)strike, // strike decimal precision is limited to 4 decimal places only
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
+        [Test, Ignore("Requires complete option data to validate chain")]
+        public void ValidateAAPLOptionChainSecurityIdentifiers()
```

Comment:
Cool! 👍

Reply:
For the record, prior to constructing this test, I tested validation for ALL option chains for ALL equities on a day around 2020-11-20. The test (exactly the same as this) passed successfully.